### PR TITLE
Report ocean CDR

### PR DESCRIPTION
**Pull request by katarkow**
Ocean alkalinity enhancement (OAE) takes up CO2 into the ocean by spreading hydrated lime (Ca(OH)2) into the ocean. The amount of CO2 taken by the ocean can be found in `vm_emiCdrTeDetail`. Producing Ca(OH)2 from limestone (CaCO3) (a process called calcination) results in CO2, which is captured with a capture rate of 90%. The process can use gas (traditional calciner) or electricity/hydrogen (electric or novel calciner). The emissions from burning gas are also captured in the CDR module (similarly to gas used for DAC).

Due to <100% capture rate, there are unavoidable emissions (from gas and calcination) which get deduced from the CDR amount already in the CDR module.
```   
vm_emiCdr(t,regi,"co2")
=e=
sum(te_used33, vm_emiCdrTeDetail(t,regi,te_used33))
+ (1 - s33_capture_rate) * (
    sum(te_ccs33, v33_co2emi_non_atm_gas(t, regi, te_ccs33))
    + sum(te_oae33, v33_co2emi_non_atm_calcination(t, regi, te_oae33))
)
```

The captured emissions (see eq below) end up either stored or released due to CCU.
```
sum(teCCS2rlf(te, rlf), vm_co2capture_cdr(t, regi, enty, enty2, te, rlf))
=e=
- vm_emiCdrTeDetail(t, regi, "dac")
+ s33_capture_rate * (
    sum(te_ccs33, v33_co2emi_non_atm_gas(t, regi, te_ccs33))
    + sum(te_oae33, v33_co2emi_non_atm_calcination(t, regi, te_oae33))
)
```
And that's where we were getting summation errors from (already for DAC only) since I split negative emissions from energy-related emissions. Here is my attempt to fix it. I know it's not perfect, but these emissions require refactoring anyway. For captured carbon from gas the CCU part of that is handled [here](https://github.com/pik-piam/remind2/blob/940c3a4f57ffb737aa78e1d28838932ab6b3625c/R/reportEmi.R#L460) and lands in `Supply`. I then add the CDR-related carbon on the supply side as well - gas to `Emi|CO2|Energy|Supply|+|Gases w/ couple prod` and calcination emissions which are basically fossil emissions to `Emi|CO2|Energy|Supply|+|Solids w/ couple prod`. This fixes the summation errors.

This PR introduces the following variables:
- Final energy for OAE:
`FE|CDR|++|OAE, electric calciner`
`FE|CDR|++|OAE, traditional calciner`
`FE|CDR|OAE, electric calciner|+|Diesel`
`FE|CDR|OAE, electric calciner|+|Electricity`
`FE|CDR|OAE, electric calciner|+|Hydrogen`
`FE|CDR|OAE, traditional calciner|+|Diesel`
`FE|CDR|OAE, traditional calciner|+|Electricity`
`FE|CDR|OAE, traditional calciner|+|Gases`
- Emissions
`Emi|CO2|CDR|OAE` - like for DACCS, these are the emissions that are the gross emissions (uptake) - unavoidable
`Emi|CO2|Cumulated|CDR|OAE`
The following are added to check if all OAE-associated CO2 adds up (they are helpful for RESCUE):
`Emi|CO2|CDR|net OAE`
`Emi|CO2|CDR|net OAE|+|Ocean Uptake`
`Emi|CO2|CDR|net OAE|+|Released emissions`
`Emi|CO2|CDR|net OAE|+|Unavoidable emissions`
`Emi|CO2|CDR|net OAE|Ocean Uptake|+|electric calciner`
`Emi|CO2|CDR|net OAE|Ocean Uptake|+|traditional calciner`
`Emi|CO2|CDR|net OAE|Calcination emissions` (this is a percentage of released + unavoidable and released + unavoidable = calcination + gas captured from energy for OAE) 
- Carbon Management
`Carbon Management|Carbon Capture|+|OAE calcination`
`Carbon Management|Storage|+|OAE calcination`
`Carbon Management|Carbon Capture|+|CDR energy ` - all the captured emissions from using gas for CDR, either DAC or OAE.

### CODE REVIEW

File: R/reportEmi.R [hunk 1]
```diff
@@ -1134,7 +1153,12 @@ reportEmi <- function(gdx, output = NULL, regionSubsetList = NULL,
                  setNames(dimSums(vm_emiMacSector[, , "co2luc"], dim = 3) * GtC_2_MtCO2,
                           "Emi|CO2|+|Land-Use Change (Mt CO2/yr)"),
                  # negative emissions from (non-BECCS) CDR (DACCS, EW)
-                 setNames((vm_emiCdrTeDetail[, , "weathering"] + vm_emiCdrTeDetail[, , "dac"] * p_share_CCS) * GtC_2_MtCO2,
+                 setNames((
+                  vm_emiCdrTeDetail[, , "weathering"]
+                  + vm_emiCdrTeDetail[, , "dac"] * p_share_CCS
+                  + (vm_emiCdrTeDetail[, , "oae_ng"] + vm_emiCdrTeDetail[, , "oae_el"])
+                    * (1 - s33_OAE_chem_decomposition * (1 - p_share_CCS))
```
Comment (strefler) [inline R/reportEmi.R:1160]: also subtract part that is never captured due to capture rate <100%, not just part that is not stored

Comment (strefler) [reply R/reportEmi.R:1160]: that part should not end up in gross emissions, but should be subtracted here

File: R/reportEmi.R [hunk 2]
```diff
@@ -1164,7 +1188,12 @@ reportEmi <- function(gdx, output = NULL, regionSubsetList = NULL,
...
                         "Emi|CO2|+|Land-Use Change (Mt CO2/yr)"),
                # negative emissions from (non-BECCS) CDR (DACCS, EW)
-               setNames((vm_emiCdrTeDetail[, , "weathering"] + vm_emiCdrTeDetail[, , "dac"] * p_share_CCS) * GtC_2_MtCO2,
+               setNames((
+                vm_emiCdrTeDetail[, , "weathering"]
+                + vm_emiCdrTeDetail[, , "dac"] * p_share_CCS
+                + (vm_emiCdrTeDetail[, , "oae_ng"] + vm_emiCdrTeDetail[, , "oae_el"])
+                  * (1 - s33_OAE_chem_decomposition * (1 - p_share_CCS))
+               ) * GtC_2_MtCO2,
                         "Emi|CO2|+|non-BECCS CDR (Mt CO2/yr)")
```
Comment (strefler) [inline R/reportEmi.R:1197]: recalculate to account only for "net CDR". Use emiCDR?

File: R/reportEmi.R [hunk 3]
```diff
@@ -1373,6 +1402,13 @@ reportEmi <- function(gdx, output = NULL, regionSubsetList = NULL,
     setNames(-vm_emiCdrTeDetail[, , "dac"] * GtC_2_MtCO2,
              "Carbon Management|Carbon Capture|+|DAC (Mt CO2/yr)"),
 
+    # total co2 captured from OAE calcination
+    setNames(
+        - s33_OAE_chem_decomposition
```
[the remark this answers is not in the document]
Comment (strefler) [reply R/reportEmi.R:1407]: might be taken care of in the calculation of vm_emiCdrTeDetail, so this would then be correct

File: R/reportEmi.R [hunk 4]
```diff
@@ -1769,6 +1810,24 @@ reportEmi <- function(gdx, output = NULL, regionSubsetList = NULL,
...
+               setNames(vm_emiCdrTeDetail[, , "oae_ng"] * GtC_2_MtCO2,
+                        "Emi|CO2|CDR|OAE|+|Ocean Uptake, traditional calciner (Mt CO2/yr)"),
+               setNames(vm_emiCdrTeDetail[, , "oae_el"] * GtC_2_MtCO2,
+                        "Emi|CO2|CDR|OAE|+|Ocean Uptake, electric calciner (Mt CO2/yr)"),
+               setNames((vm_emiCdrTeDetail[, , "oae_ng"] + vm_emiCdrTeDetail[, , "oae_el"]) * GtC_2_MtCO2,
+                        "Emi|CO2|CDR|OAE|+|Ocean Uptake (Mt CO2/yr)"),
+               # OAE process emissions
+               setNames(out[, , "Carbon Management|Carbon Capture|+|OAE calcination (Mt CO2/yr)"]
+                        - out[, , "Carbon Management|Storage|+|OAE calcination (Mt CO2/yr)"],
+                        "Emi|CO2|CDR|OAE|+|Calcination emissions (Mt CO2/yr)"),
```
Comment (strefler) [inline R/reportEmi.R:1824]: add emissions that are not captured

File: R/reportEmi.R [hunk 5]
```diff
@@ -1769,6 +1810,24 @@ reportEmi <- function(gdx, output = NULL, regionSubsetList = NULL,
                # total DACCS
                setNames(-out[, , "Carbon Management|Storage|+|DAC (Mt CO2/yr)"],
                         "Emi|CO2|CDR|DACCS (Mt CO2/yr)"),
+
+               # gross OAE
+               setNames(vm_emiCdrTeDetail[, , "oae_ng"] * GtC_2_MtCO2,
+                        "Emi|CO2|CDR|OAE|+|Ocean Uptake, traditional calciner (Mt CO2/yr)"),
+               setNames(vm_emiCdrTeDetail[, , "oae_el"] * GtC_2_MtCO2,
```
Comment (strefler) [inline R/reportEmi.R:1817]: these two should also get a "+" instead of the "," so they add up to "Ocean Uptake"

File: R/reportEmi.R [hunk 6]
```diff
@@ -1813,6 +1873,18 @@ reportEmi <- function(gdx, output = NULL, regionSubsetList = NULL,
                  # total DACCS
                  setNames(-out[, , "Carbon Management|Storage|+|DAC (Mt CO2/yr)"],
                           "Emi|CO2|CDR|DACCS (Mt CO2/yr)"),
+                 # gross OAE
+                 setNames((vm_emiCdrTeDetail[, , "oae_ng"] + vm_emiCdrTeDetail[, , "oae_el"] ) * GtC_2_MtCO2,
+                           "Emi|CO2|CDR|OAE|+|Ocean Uptake (Mt CO2/yr)"),
+                 # OAE process emissions
+                 setNames(out[, , "Carbon Management|Carbon Capture|+|OAE calcination (Mt CO2/yr)"]
+                           - out[, , "Carbon Management|Storage|+|OAE calcination (Mt CO2/yr)"],
```
Comment (strefler) [inline R/reportEmi.R:1881]: add emissions that are never captured here as well

File: inst/compareScenarios/cs_09_carbon_management.Rmd [hunk 7]
```diff
@@ -78,7 +78,6 @@ showLinePlots(data, "Carbon Management|Storage|Share of annual potential used")
 showAreaAndBarPlots(data,
                     vars = c("Carbon Management|Storage"),
                     tot = "Carbon Management|Storage|Maximum annual CO2 storage potential")
-```
```
Comment (strefler) [inline inst/compareScenarios/cs_09_carbon_management.Rmd:81]: add gross OAE, emissions, and net OAE to check if it adds up

File: R/reportEmi.R (hunk 2 above)
Comment (katarkow) [reply R/reportEmi.R:1197]: If I understand correctly, this variable should be negative emissions, so gross(ish) = gross - released CO2 that is unavoidable due to <100% capture rate. So it should be `vm_emiCdr - vm_emiCdrTeDetail[, , "dac"] * (1 - p_share_CCS)`

E.g.: `Emi|GHG = Emi|CO2|+|Energy (952.393059421011) + Emi|CO2|+|Industrial Processes (18.70015799332) + Emi|CO2|+|Land-Use Change (38.7566666666702) + Emi|CO2|+|non-BECCS CDR (0) + Emi|GHG|+|CH4 (352.029277806816) + Emi|GHG|+|F-Gases (27.202629) + Emi|GHG|+|N2O (76.9251374089826)`

File: R/reportEmi.R (hunk 1 above)
Comment (katarkow) [reply R/reportEmi.R:1160]: It is now deducted from vm_emiCdr

File: R/reportEmi.R (hunk 4 above)
Comment (katarkow) [reply R/reportEmi.R:1824]: I created a new variable for this: 'Emi|CO2|CDR|net OAE|+|Unavoidable emissions (Mt CO2/yr)'

**Review by strefler — APPROVED** [review]

**Pull request merged**